ci/publish_github_release.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# NOTE: please do not set trace mode (-x)
+# as it will leak credentials in the logs.
+set -e
+
+if [[ ! -f "${ENVOY_SRCDIR}/build_release_stripped/envoy" ]]; then
+    echo "could not locate envoy binary at path: ${ENVOY_SRCDIR}/build_release_stripped/envoy"
+    exit 1
+fi
+
+if [[ -z "${GITHUB_TOKEN:-}" ]]; then
+    echo "environment variable GITHUB_TOKEN unset; cannot continue with publishing."
+    exit 1
+fi
+
+if [[ -z "${GITHUB_USER:-}" ]]; then
+    echo "environment variable GITHUB_USER unset; cannot continue with publishing."
+    exit 1
+fi
+
+if [[ -z "${GITHUB_REPO:-}" ]]; then
+    echo "environment variable GITHUB_REPO unset; cannot continue with publishing."
+    exit 1
+fi
+
+if [[ -z "${CIRCLE_TAG:-}" ]]; then
```

Comment:
I'd move this to be the very first check since this file runs on every build. We should bail early if it's not a tag event rather than failing on unnecessary dependencies